queries.sql

```diff
@@ -14,4 +14,126 @@ SELECT * FROM animals WHERE neutered = TRUE; --Find all animals that are neutere
 
 SELECT * FROM animals WHERE name != 'Gabumon'; --Find all animals not named Gabumon.
 
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3; --Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3; --Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
+
+/* 
+Inside a transaction update the animals table 
+by setting the species column to unspecified. 
+Verify that change was made. Then roll back 
+the change and verify that the species 
+columns went back to 
+the state before the transaction.
+*/
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+
+ROLLBACK;
+
+/*
+Update the animals table by setting the 
+species column to digimon for all 
+animals that have a name ending in mon.
+*/
+
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+/*
+Update the animals table by setting 
+the species column to pokemon for all 
+animals that don't have species already set.
+*/
+BEGIN;
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+/*
+Inside a transaction delete all records in the animals table, 
+then roll back the transaction.
+*/
+BEGIN;
+
+DELETE FROM animals;
+ROLLBACK;
+
+-- Delete all animals born after Jan 1st, 2022.
+
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT DELETEDATE2022;
+
+
+-- Update all animals' weight to be their weight multiplied by -1.
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+-- Rollback to the savepoint
+ROLLBACK TO DELETEDATE2022;
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+-- Commit transaction
+COMMIT;
+
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals
+WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+
+-- Sum of escape attempts by neutered
+SELECT neutered, SUM(neutered) FROM animals
```

Comment:
You did well, kindly note that the who escapes the most, neutered and non-neutered should be  AVG(escape_attempts) not  SUM(neutered) 😄